### ISSUE TITLE
XMPP authentication failed

### ISSUE THREAD

**Issue by pokal4u** [node-xmpp/node-xmpp#279]
Hi,
I am new to node js xmpp
I try to implement chat application using node-xmpp.
I getting XMPP authentication failed eror.

My code:

```javascript
var express = require('express');
var stormpath = require('express-stormpath');
var xmpp = require('node-xmpp');
var app = express();

app.set('views', './views');
app.set('view engine', 'jade');
app.get('/', function(req, res) {
  res.render('home', {
    title: 'Welcome'
  });
});
app.use('/profile', function(req, res) {
  res.render('profile', {
    title: 'Welcome'
  });
});
var client = new xmpp.Client({jid: 'user@gmail.com', password: '123456'});

client.on('online', function(data) {
    console.log('Connected as ');
    
});

client.on('stanza', function(stanza) {
    console.log('RECEIVED STANZA', stanza);
});

client.on('offline', function () {
    console.log('Client is offline');
});

client.on('connect', function () {
    console.log('Client is connected');
});

client.on('error', function(data){
    console.log(data);
});

client.connect();
app.listen(8088);
```

when run "node app.js" getting response like:

```
Client is connected
Client is connected
XMPP authentication failure
Client is offline
Client is offline
```

Please help me...

Thanks
Pokal

**Comment by pokal4u** [comment]
Please help me..

**Comment by pokal4u** [comment]
HI,

Now I am sending messages from admin@localhost to test@localhost

My code:

var xmpp = require('node-xmpp');
var c2s = new xmpp.C2SServer({
    port: 8088,
    domain: 'localhost'
});
// On Connect event. When a client connects.
c2s.on("connect", function() {
    c2s.on("register", function(opts, cb) {
        console.log("REGISTER");
		cb(true);
    });	
	client = new xmpp.Client({jid: 'admin@localhost', password: 'admin123'});

    client.on("authenticate", function(opts, cb) {
        console.log("AUTH" + opts.jid + " -> " +opts.password); 
		cb(null, opts); 
    });
	
	client.on('error', function(err) {
		console.log(err.toString());
	});

	client.on('online', function() {
		console.log("online");
		var stanza = new xmpp.Element('message', { to: 'test@localhost', type: 'chat', 'xml:lang': 'ko' }).c('body').t('test');
		client.send(stanza);
	});
	
    client.on("stanza", function(stanza) {
        console.log("STANZA" + stanza);
    });
    client.on("disconnect", function(client) {
        console.log("DISCONNECT");
    });
});

 I saw those messages in ejabbered admin

![xmpp](https://cloud.githubusercontent.com/assets/498771/8643499/44639e8e-2952-11e5-9b44-b5353d3d1f27.png)

How to print those messages?

Thanks
Pokal

**Comment by lloydwatkin** [comment]
Do you mean how would you receive those messages?

You'd need to connect as test@localhost and send a presence to the server, it will then forward the offline messages to you.

Please raise any future client issues against `node-xmpp-client` as `node-xmpp` is deprecated.

**Comment by pokal4u** [comment]
Yes.
How could send a presence to the server?

any example code...

**Comment by lloydwatkin** [comment]
I suggest you look at the related RFC http://xmpp.org/rfcs/rfc6121.html

**Comment by pokal4u** [comment]
HI,  I am getting error like this

![error](https://cloud.githubusercontent.com/assets/498771/8650048/7e419594-298d-11e5-8056-7af5c0cf7b1f.png)

**Comment by pokal4u** [comment]
Hi,
When I send message using below code:
var stanza = new xmpp.Element('message', { to: 'admin@localhost', type: 'chat' }).c('body').t('test');
client.send(stanza);

for retrieving?

Thanks

**Comment by lloydwatkin** [comment]
I'm afraid I don't understand the question. Please could you rephrase and post on the `node-xmpp-client` repository instead please?
